homeassistant/components/tesla/__init__.py

```diff
@@ -171,54 +205,55 @@ async def async_setup_entry(hass, config_entry):
 
 async def async_unload_entry(hass, config_entry) -> bool:
     """Unload a config entry."""
-    await asyncio.gather(
-        *[
-            hass.config_entries.async_forward_entry_unload(config_entry, component)
-            for component in TESLA_COMPONENTS
-        ]
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(config_entry, component)
+                for component in TESLA_COMPONENTS
+            ]
+        )
     )
     for listener in hass.data[DOMAIN][config_entry.entry_id][DATA_LISTENER]:
         listener()
     username = config_entry.title
-    hass.data[DOMAIN].pop(config_entry.entry_id)
-    _LOGGER.debug("Unloaded entry for %s", username)
-    return True
+    if unload_ok:
+        hass.data[DOMAIN].pop(config_entry.entry_id)
+        _LOGGER.debug("Unloaded entry for %s", username)
+        return True
+    return False
 
 
 async def update_listener(hass, config_entry):
     """Update when config_entry options update."""
     controller = hass.data[DOMAIN][config_entry.entry_id]["controller"]
     old_update_interval = controller.update_interval
     controller.update_interval = config_entry.options.get(CONF_SCAN_INTERVAL)
-    _LOGGER.debug(
-        "Changing scan_interval from %s to %s",
-        old_update_interval,
-        controller.update_interval,
-    )
+    if old_update_interval != controller.update_interval:
+        _LOGGER.debug(
+            "Changing scan_interval from %s to %s",
+            old_update_interval,
+            controller.update_interval,
+        )
 
 
 class TeslaDevice(Entity):
     """Representation of a Tesla device."""
 
-    def __init__(self, tesla_device, controller, config_entry):
+    def __init__(self, tesla_device, coordinator):
         """Initialise the Tesla device."""
         self.tesla_device = tesla_device
-        self.controller = controller
-        self.config_entry = config_entry
-        self._name = self.tesla_device.name
-        self.tesla_id = slugify(self.tesla_device.uniq_name)
-        self._attributes = {}
-        self._icon = ICONS.get(self.tesla_device.type)
+        self.coordinator = coordinator
+        self._attributes = self.tesla_device.attrs.copy()
 
     @property
     def name(self):
         """Return the name of the device."""
-        return self._name
+        return self.tesla_device.name
 
     @property
     def unique_id(self) -> str:
         """Return a unique ID."""
-        return self.tesla_id
+        return slugify(self.tesla_device.uniq_name)
```

Comment:
i believe slugify can be slow but also based on this previous review I dont think its needed.

https://github.com/home-assistant/core/pull/37581#discussion_r450628019

Reply:
actually since it was used previously it would be a breaking change so it should stay i guess

Reply:
We should not use slugifu on every state update. That's not needed. It's ok to use it once when creating the entity if it's really needed.

Reply:
Reverted in #38680